Riot/Managers/UserSessions/UserSessionsService.swift

```diff
@@ -0,0 +1,131 @@
+// 
+// Copyright 2021 New Vector Ltd
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+
+import Foundation
+
+// MARK: - UserSessionsService notification constants
+extension UserSessionsService {
+    public static let didAddUserSession = Notification.Name("UserSessionsServiceDidAddUserSession")
+    public static let willRemoveUserSession = Notification.Name("UserSessionsServiceWillRemoveUserSession")
+    public static let didRemoveUserSession = Notification.Name("UserSessionsServiceDidRemoveUserSession")
+    public static let userSessionDidChange = Notification.Name("UserSessionsServiceUserSessionDidChange")
+    
+    public struct NotificationUserInfoKey {
+        static let userSession = "userSession"
+        static let userId = "userId"
+    }
+}
+
+/// UserSessionsService enables to manage multiple user sessions and all logic around sessions management.
+/// TODO: Move MXSession and MXKAccountManager code from LegacyAppDelegate to this place. Create a UserSessionService to make per session management if needed.
+@objcMembers
+class UserSessionsService: NSObject {
+    
+    // MARK: - Properties
+    
+    // MARK: Private
+    
+    private(set) var userSessions: [UserSession] = []
```

Comment:
We could have self.userSessions public. It will useful in some existing places.

Reply:
The set is private but the get is public